### PR TITLE
[GSoD2020 Branch] Fix CSS Styles and Layouts

### LINKED ISSUE #475: CSS styles for the new version (GSoD2020) of the Interactive Book
## Describe the bug

The new version of the book (being developed under the GSoD 2020 project) replaced the old version Just-the-docs (JTD) Jekyll theme, which was installed locally and included styling and structure hardcoded into it, with the current upstream JTD version installed as a Gem/plugin for Jekyll. In the process some of the original style of the book was lost, either because it was hardcoded in the template, or because it was closely related to document structures which are not now part of theme.

At least, the following problems with the style have been identified (there might be more). Check the attached screenshot for details (left is the pre-GSoD2020 version, right is the new version):

1. In the navigation bar, single pages at the top level and chapters (group of pages) have different "indentation" ("yellowish" 1 in the screenshot).
2. In the navbar, the distance from the bullet marker to the text is larger than in the original design (red 2 in the screenshot).
3. Chapter title is located more to the right than the sections in the chapter. Also, the section title style seems be a bit of with respect to the rest of the nav bar: i.e. color, marker, etc., although it is the same in the current version's style. (blue 3 in the screenshot).
4. Links at the top right are no longer buttons (they are part of the JTD theme) but they can be styled as such, it just need the proper CSS code (red 4 in the screenshot).
5. Table of content headline doesn't match the original book style. (Blue 5 in the screenshot).

## Expected behavior

Consistent style and closer to the current one

## Screenshots

CVIB_versus-2020-10-07_labels

## Additional context

All css should go in the `_sass/custom/custom.scss` as indicated in the JTD [documentation](https://pmarsceill.github.io/just-the-docs/docs/customization/#override-and-completely-custom-styles)

See PR #474

### LINKED ISSUE #483: Table Contents not Visible in Dark Mode [GSoD 2020 Branch]
## Describe the bug

The table contents in the new Interactive Book ( GSoD 2020 Branch ) are only visible in light mode. While changing to Dark Mode the contents vanish away and only an empty table is visible.
Please find the screenshots attached below

## To Reproduce

1.  Go to 'betalearn.circuitverse.org'
2.  Click on 'Any Page Containig Table'
3.  Click on 'Dark Mode'
4.  See error

## Expected behavior

The table should show contents even in dark mode.

## Screenshots

Light Mode:
image

Dark Mode:
image

## Desktop (please complete the following information):

* OS: [Windows]
* Browser [Chrome]

## Additional context

Please make PRs on GSOD-2020 Branch

### PR DESCRIPTION
**Pull request by manjotsidhu**
This Pull Request fixes all the CSS related styling issues in GSoD2020 branch and Layout Enhancements.
I have made this all-in-one pull request because there are contributions done by others in bits and pieces and that too old. 

Fixes #475, Fixes #483 
Includes #488 (by @anushri20)

#### Changes done:
- [x] Fix Table of Contents title style (by @anushri20)
- [x] CV flavored Auxiliary Navigation Buttons (by @anushri20) 
- [x] Fix Buttons Bar, uneven Sidebar size due to overflow. 
- [x] Fix hover (feedback) color in Dark mode.
- [x] CV Flavored side navigation 
- [x] Fix CSS Style for table background in Dark Mode.
- [x] New Home Layout which shows header as well in Home Page. 

#### Preview Link(s): 
https://unruffled-sammet-5acbc2.netlify.app/

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add the link(s) for all the pages changed/updated from the checks tab after checks complete)
- [x] Tried Squashing the commits into one.

### CODE REVIEW

**Comment by manjotsidhu** [comment]
@Shivansh2407 Kindly verify the layout of the top right auxiliary buttons. I have intentionally not applied the button style to mobile because button style looks weird at that place and wanted to keep that minimal.

https://user-images.githubusercontent.com/22657113/108687768-1e6ea080-751d-11eb-9742-2b617107d6c2.mp4

File: _sass/custom/custom.scss [hunk 1]
```diff
@@ -264,21 +270,65 @@ $text-color: #111111;
...
+  .nav-list-item.active {
+    list-style: disc url("/assets/images/nav-marker-active.svg") inside;
+  }
+}
+
+// Nested nav lists
+.nav-list .nav-list-item > .nav-list .nav-list-item .nav-list-link {
+  padding-left: 1rem;
+
+  &.active {
```
[the remark this answers is not in the document]
Comment (manjotsidhu) [reply _sass/custom/custom.scss:292]: Finding nested list items can't be reduced to have depth of applicability of 2

**Comment by manjotsidhu** [comment]
> @Shivansh2407 Kindly verify the layout of the top right auxiliary buttons. I have intentionally not applied the button style to mobile because button style looks weird at that place and wanted to keep that minimal.
> 
>  2021-02-22.14-27-18.mp4

@Shivansh2407 @tachyons

File: _sass/custom/custom.scss [hunk 2]
```diff
@@ -264,21 +270,65 @@ $text-color: #111111;
...
+      padding-left: 0;
+  }
+
+  .nav-list-item.active {
+    list-style: disc url("/assets/images/nav-marker-active.svg") inside;
+  }
+}
+
+// Nested nav lists
+.nav-list .nav-list-item > .nav-list .nav-list-item .nav-list-link {
```
[the remark this answers is not in the document]
Comment (manjotsidhu) [reply _sass/custom/custom.scss:289]: Finding nested list items can't be reduced to have depth of applicability of 2

**Comment by tachyons** [comment]
Sorry, accidentally merged, adding css class in every files doesn't seem a good approach

**Comment by manjotsidhu** [comment]
> Sorry, accidentally merged, adding css class in every files doesn't seem a good approach

If you are talking about using `text-delta` class, I was considering the same before implementing but found that the approach is already being followed by master branch and partially used in other pages.

**Comment by manjotsidhu** [comment]
Reopened this PR #528

**Pull request merged**